commit a85b4deb7d6e661c741bdcf42fab1f8aef217e4e (refs/changes/43/2143/1)
Author: Jordan Hendricks <jordan.hendricks@joyent.com>
Date:   2017-06-19T02:27:07+00:00 (2 years, 4 months ago)
    
    MANTA-3296 Mako returned different md5 sums for the same MPU object
    Reviewed By:
    Approved By:

diff --git a/src/http/modules/mpu/ngx_http_mpu_commit_module.c b/src/http/modules/mpu/ngx_http_mpu_commit_module.c
index a682babf..0e077728 100644
--- a/src/http/modules/mpu/ngx_http_mpu_commit_module.c
+++ b/src/http/modules/mpu/ngx_http_mpu_commit_module.c
@@ -601,6 +601,13 @@ mpu_convert_md5(mpu_request_t *mpcr)
 static boolean_t
 mpu_determine_output_md5(mpu_request_t *mpcr, int fd)
 {
+    /*
+     * It's possible we have already started calculating the md5 sum of the
+     * object before this function is called, so we need to reinitialize the
+     * state of the md5 sum before starting from the beginning of the object.
+     */
+	ngx_md5_init(&mpcr->mpcr_md5);
+
 	for (;;) {
 		ssize_t ret;
 
